samples/snippets/webhook_log_session_info.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2022, Google LLC
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" DialogFlow CX: webhook to log session info for each request."""
+
+# [START dialogflow_cx_v3_webhook_log_session_info]
+
+import re
+
+
+def log_session_info_for_troubleshooting(request):
+    """Webhook will log session id corresponding to request."""
+
+    req = request.get_json()
```

Comment:
If there's a concern about unhandled exceptions, you might want to wrap this in a try block in case the request does not contain well-formed JSON.

Reply:
There shouldn't be a concern of incorrect json here. Since the [doc](https://cloud.google.com/dialogflow/cx/docs/concept/webhook#webhook-request) indicates that the request will contain JSON data this behavior should be guaranteed.